rerun_py/rerun_sdk/rerun/archetypes/image_ext.py

```diff
@@ -302,27 +330,8 @@ def compress(self: Any, jpeg_quality: int = 95) -> EncodedImage | Image:
                     f"Cannot JPEG compress an image of datatype {image_format.channel_datatype}. Only U8 is supported.",
                 )
 
-            buf = None
-            if self.buffer is not None:
-                buf = self.buffer.as_arrow_array().values.to_numpy().view(image_format.channel_datatype.to_np_dtype())
-
-            if buf is None:
-                raise ValueError("Cannot JPEG compress an image without data")
-
-            # Note: np array shape is always (height, width, channels)
-            if image_format.color_model == ColorModel.L:
-                image = buf.reshape(image_format.height, image_format.width)
-            else:
-                image = buf.reshape(image_format.height, image_format.width, 3)
-
-            # PIL doesn't understand BGR.
-            if image_format.color_model == ColorModel.BGR:
-                mode = "RGB"
-                image = image[:, :, ::-1]
-            else:
-                mode = str(image_format.color_model)
+            pil_image = self.as_pil_image()
 
-            pil_image = PILImage.fromarray(image).convert(mode)
```

Comment:
This `.convert(mode)` produced a warning. I decided to refactor the code while I was at it